building-with-ocean/asset-hosting.md

```diff
@@ -1,51 +1,142 @@
+description: Tutorial to publish assets using hosting services like AWS and Azure.
 ---
 
 # Publishing with Hosting Services
 
 ### Overview
 
-To publish assets on the Ocean Marketplace, publishers must provide a link(an URL) to the file. It is up to the asset publisher to decide where to host the asset. For example, a publisher can store the content on their Google Drive, AWS server, private cloud server, or other third-party hosting services. Through publishing, the URL of the asset is encrypted and stored as a part of DDO on the blockchain. Buyers don't have access directly to the URL, but they interact with the Provider, which decrypts the URL and acts as a proxy to serve the asset. The DDO only stores the location of the file, which is accessed on-demand by the Provider. Implementing a security policy that allows only the Provider to access the URL and blocks requests from other unauthorized actors is recommended. One of the possible ways to achieve this is to allow only the Provider's IP address to access the URL. But, not all hosting services provide this feature. So, the publishers must consider the security features while choosing a hosting service.
+To publish assets on the Ocean Marketplace, publishers must provide a link(an URL) to the file. It is up to the asset publisher to decide where to host the asset. For example, a publisher can store the content on their AWS server, private cloud server, or other third-party hosting services. Through publishing, the URL of the asset is encrypted and stored as a part of DDO on the blockchain. Buyers don't have access directly to the URL, but they interact with the Provider, which decrypts the URL and acts as a proxy to serve the asset. The DDO only stores the location of the file, which is accessed on-demand by the Provider. Implementing a security policy that allows only the Provider to access the URL and blocks requests from other unauthorized actors is recommended. One of the possible ways to achieve this is to allow only the Provider's IP address to access the URL. But, not all hosting services provide this feature. So, the publishers must consider the security features while choosing a hosting service.
 
-On Ocean Marketplace, a publisher must provide the link to the asset during publish step. Once the asset is published, this link cannot be changed. So, it is essential that the publisher correctly sets this field (shown in the below image).
+On Ocean Marketplace, a publisher must provide the link to the asset during publish step in the field shown in the below image.
 
 ![Publish - File URL field](../.gitbook/assets/marketplace-publish-file-field.png)
 
 ### Hosting services
 
 Publishers can choose any hosting service of their choice. The below section explains how to use commonly used hosting services with Ocean Marketplace.
 
-#### Google Drive
+As a mention, Google Drive is not a reliable data storage solution that integrates well with Ocean.
```

Comment:
Maybe change this to:
"Please use a proper hosting solution to keep your files. When using other systems(like Google Drive) that are not specifically designed for this use case, you risk not gaining traction. Systems like Google Drive, include various virus checks and rate limiters that prevent the `Provider` to download the asset once it was purchased.